.github/actions/setup-env/action.yml

```diff
@@ -30,33 +24,33 @@ outputs:
 runs:
   using: composite
   steps:
-    - name: Set up python ${{ inputs.python-version }}
+    - name: Set up pdm with python ${{ inputs.python-version }}
       id: setup-python
-      uses: actions/setup-python@v4
+      uses: pdm-project/setup-pdm@v3
       with:
         python-version: ${{ inputs.python-version }}
-        cache: "pip"
+        version: '2.2'
+        cache: true
         cache-dependency-path: ${{ inputs.cache-dependency-path }}
+        enable-pep582: false # Enable PEP 582 package loading globally
+
+    - name: Ignore saved pythons
+      shell: bash
+      run: |
+        echo "PDM_IGNORE_SAVED_PYTHON=1" >> $GITHUB_ENV
 
     - name: Update pip, wheel, setuptools
       shell: bash
       run: pip install -U pip wheel setuptools
 
     - name: Install dependencies
-      if: inputs.nox-setup || inputs.dependencies
+      if: inputs.pdm-command
       env:
-        NOX_SETUP: ${{ inputs.nox-setup }}
-        DEPS: ${{ inputs.dependencies }}
+        PDM_COMMAND: ${{ inputs.pdm-command }}
       shell: bash
       run: |
-        [ -n "$NOX_SETUP" ] && DEPS="$DEPS nox"
-        python -m pip install $DEPS
-
-        if [ "*" == "$NOX_SETUP" ]; then
-          nox -s setup
-        elif [ -n "$NOX_SETUP" ]; then
-          nox -s setup -- $NOX_SETUP
-        fi
+        pip install nox
+        $PDM_COMMAND
```

Comment:
I left pdm install in all of the inputs intentionally in the event I want to run a different command, like change a config setting or something.

Reply:
In that case, why make it an input to the composite action? It could just be a separate step in the main workflow, since it's not needed for most things (`nox` already installs things in venvs, no need to make pdm install them globally first).

Reply:
Refactored in https://github.com/DisnakeDev/disnake/pull/836/commits/05446c443f2e72ab7aff518f95b80de9f2193ec6.